CRM/Contribute/Import/Parser/Contribution.php

```diff
@@ -833,15 +808,8 @@ private function deprecatedFormatParams($params, &$values, $create = FALSE) {
             $values['pledge_payment_id'] = $pledgePaymentDetails['id'];
           }
           else {
-            return civicrm_api3_create_error('Contribution and Pledge Payment amount mismatch for this record. Contribution row was skipped.');
-          }
-          break;
-
-        case 'contribution_campaign_id':
-          if (empty(CRM_Core_DAO::getFieldValue('CRM_Campaign_DAO_Campaign', $params['contribution_campaign_id']))) {
-            return civicrm_api3_create_error('Invalid Campaign ID provided. Contribution row was skipped.');
+            throw new CRM_Core_Exception('Contribution and Pledge Payment amount mismatch for this record. Contribution row was skipped.', CRM_Import_Parser::ERROR);
           }
-          $values['contribution_campaign_id'] = $params['contribution_campaign_id'];
```

Comment:
@eileenmcnaughton is there a reason for removing the campaign id validation here?

Reply:
yep - because it is already done in the `validate` stage (so is the `email` validation now I look - that should come out too)